### PR TITLE
CLDC-2665 Switch lettings ppcodenk

### PR DESCRIPTION
**Pull request by ghost**
Done:

- refactor and simplify Question model (no need to handle lettings and sales logs differently)
- swap 0 and 1 `ppcodenk` values in code

Partially done:
- updated tests to reflect changes. In the importers there's something odd going because of the way StandardError is rescued. You'll end up in a recursive loop that complicates debugging, my best guess is somewhere I haven't flipped the `ppcodenk` values.  Try running to see`rspec ./spec/services/imports/lettings_logs_import_service_spec.rb:191` 
- update fixtures, some might still need updating

### CODE REVIEW

File: app/models/lettings_log.rb [hunk 1]
```diff
@@ -260,8 +260,8 @@ def postcode_known?
   end
 
   def previous_postcode_known?
-    # 1: Yes
-    ppcodenk == 1
+    # 0: Yes
+    postcode_known&.zero?
```
Comment (kosiakkatrina) [inline app/models/lettings_log.rb]: Is this meant to be previous postcode?

File: app/services/bulk_upload/lettings/year2022/row_parser.rb [hunk 2]
```diff
@@ -1283,12 +1283,7 @@ def previous_la_known
   end
 
   def ppcodenk
-    case field_65
-    when 1
-      1
-    when 2
-      0
-    end
+    field_65 == 1 ? 0 : 1
```
Comment (kosiakkatrina) [inline app/services/bulk_upload/lettings/year2022/row_parser.rb]: should this be a required field tho?

File: app/models/lettings_log.rb (hunk 1 above)
Comment (ghost) [reply app/models/lettings_log.rb]: yes good catch, fixing

File: app/services/bulk_upload/lettings/year2022/row_parser.rb (hunk 2 above)
Comment (ghost) [reply app/services/bulk_upload/lettings/year2022/row_parser.rb]: I think so? Just inverting 0 and 1 in this change, do you think it needs something else?

Comment (kosiakkatrina) [reply app/services/bulk_upload/lettings/year2022/row_parser.rb]: Checked the values for this and updated!

File: app/models/form/question.rb [hunk 3]
```diff
@@ -166,28 +166,17 @@ def label_from_value(value, _log = nil, _user = nil)
...
       value == 1
     when "radio"
-      is_lettings ? RADIO_YES_VALUE_LETTINGS[id.to_sym]&.include?(value) : RADIO_YES_VALUE_SALES[id.to_sym]&.include?(value)
+      RADIO_YES_VALUE[id.to_sym]&.include?(value)
     else
       %w[yes].include?(value.downcase)
     end
   end
 
-  def value_is_no?(value, is_lettings)
```
Comment (natdeanlewissoftwire) [inline app/models/form/question.rb:180]: I assume we don't use this anywhere?

File: app/models/form/question.rb [hunk 4]
```diff
@@ -338,6 +327,7 @@ def inferred_answer_value(log)
     is_carehome: [1],
     hbrentshortfall: [1],
     net_income_value_check: [0],
+    ppcodenk: [0],
   }.freeze
 
   RADIO_NO_VALUE = {
```
Comment (natdeanlewissoftwire) [inline app/models/form/question.rb:344]: Do we need this if we don't use value_is_no?

**Review by natdeanlewissoftwire — APPROVED** [review]
lgtm if you're happy you've caught everywhere this change needs to happen :)

File: app/models/form/question.rb (hunk 3 above)
Comment (kosiakkatrina) [reply app/models/form/question.rb:180]: Doesn't look like it

File: app/models/form/question.rb (hunk 4 above)
Comment (kosiakkatrina) [reply app/models/form/question.rb:344]: Uhm, good point. I think I might bring back value_is_no? Cause we've got those other convenience methods

**Pull request merged**